tests/acceptance/gcp-secrets.bats

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env bats
+
+#load _helpers
+#
+#SKIP_TEARDOWN=true
+VAULT_ADDR='http://127.0.0.1:8200'
+VAULT_IMAGE="${VAULT_IMAGE:-hashicorp/vault:1.9.0-rc1}"
+
+if [[ -z SERVICE_ACCOUNT_ID ]]
+then
+    echo "SERVICE_ACCOUNT_ID env is not set. Exiting.."
+    exit 1
+fi
+
+if [[ -z PATH_TO_CREDS ]]
+then
+    echo "PATH_TO_CREDS env is not set. Exiting.."
+    exit 1
+fi
+
+if [[ -z GOOGLE_PROJECT ]]
+then
+    echo "GOOGLE_PROJECT env is not set. Exiting.."
+    exit 1
+fi
+
+if [[ -z GOOGLE_APPLICATION_CREDENTIALS ]]
+then
+    echo "GOOGLE_APPLICATION_CREDENTIALS env is not set. Exiting.."
+    exit 1
+fi
+
+export SETUP_TEARDOWN_OUTFILE=/tmp/output.log
+
+setup(){
+    { # Braces used to redirect all setup logs.
+    # 1. Configure Vault.
+    VAULT_TOKEN='root'
+    DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+    docker pull ${VAULT_IMAGE?}
+
+    docker run \
+      --name=vault \
+      --hostname=vault \
+      -p 8200:8200 \
+      -e VAULT_DEV_ROOT_TOKEN_ID="root" \
+      -e VAULT_ADDR="http://localhost:8200" \
+      -e VAULT_DEV_LISTEN_ADDRESS="0.0.0.0:8200" \
+      --privileged \
+      --detach ${VAULT_IMAGE?}
+
+    # Replace with a check
+    sleep 2
+
+    vault login ${VAULT_TOKEN?}
+
+    vault secrets enable gcp
+    } >> $SETUP_TEARDOWN_OUTFILE
+}
+
+teardown(){
+    if [[ -n $SKIP_TEARDOWN ]]; then
+        echo "Skipping teardown"
+        return
+    fi
+
+    { # Braces used to redirect all teardown logs.
+
+    vault secrets disable gcp
+    # If the test failed, print some debug output
+    if [[ "$BATS_ERROR_STATUS" -ne 0 ]]; then
+        docker logs vault
```

Comment:
Just a note here. When going through this and trying to debug issues I didn't find the vault logs to be useful. What I did find useful was adding this line to the `teardown`

```
echo "${BATS_TEST_NAME}: [$BATS_ERROR_STATUS]: ${output}" >&2
```

This is redirected to stderr so the `if [[ "$BATS_ERROR_STATUS" -ne 0 ]]; then` check is not needed.

Reply:
To clarify a bit, I am not saying the vault logs will never be useful. Maybe we could send them to the output.log file and output only the actual error when the test fails.

Reply:
I like the addition to stderr, so I included that line in teardown and kept the writing of the logs to a temp output file in case we ever end up needing those